bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/BigtableDataSettingsFactory.java

```diff
@@ -0,0 +1,174 @@
+ * {@link BigtableDataSettings} instance.
+ */
+public class BigtableDataSettingsFactory {
+  /** Constant <code>LOG</code> */
+  private static final Logger LOG = new Logger(BigtableDataSettingsFactory.class);
+
+  /**
+   * <p>
+   * fromBigtableOptions.
+   * </p>
+   * @param configuration a {@link Configuration} object.
+   * @return a {@link BigtableDataSettings} object.
+   * @throws IOException if any.
+   */
+  public static BigtableDataSettings fromBigtableOptions(final BigtableOptions options)
+      throws IOException, GeneralSecurityException {
+    BigtableDataSettings.Builder builder = BigtableDataSettings.newBuilder();
+
+    InstanceName instanceName = InstanceName.newBuilder().setProject(options.getProjectId())
+        .setInstance(options.getInstanceId()).build();
+    builder.setInstanceName(instanceName);
+    builder.setAppProfileId(options.getAppProfileId());
+
+    LOG.debug("endpoint host %s.", options.getDataHost());
+    LOG.debug("endpoint host %s.", options.getPort());
+    builder.setEndpoint(options.getDataHost() + ":" + options.getPort());
+
+    createCredentialProvider(builder, options.getCredentialOptions());
+
+    setBulkOptions(builder, options.getBulkOptions());
+
+    RetryOptions retryOptions = options.getRetryOptions();
+    if (retryOptions.enableRetries()) {
+      // Creating retrySetting from retryOptions as well callOptionsConfig
+      RetrySettings retrySettings = RetrySettings.newBuilder()
+          .setInitialRetryDelay(ofMillis(retryOptions.getInitialBackoffMillis()))
+          .setRetryDelayMultiplier(retryOptions.getBackoffMultiplier())
+          .setMaxRetryDelay(ofMillis(retryOptions.getMaxElapsedBackoffMillis()))
+          .setMaxAttempts(retryOptions.getMaxScanTimeoutRetries())
+          .setInitialRpcTimeout(
+            ofMillis(options.getCallOptionsConfig().getShortRpcTimeoutMs()))
+          .setMaxRpcTimeout(ofMillis(options.getCallOptionsConfig().getLongRpcTimeoutMs()))
+          .build();
+      //TODO rahulkql:  Is this correct??, as these settings are for BigtableOptions 
+      //whole but here we are just setting it for bulkMutation.
+      builder.bulkMutationsSettings().setRetrySettings(retrySettings);
+      builder.checkAndMutateRowSettings().setRetrySettings(retrySettings);
+      builder.readModifyWriteRowSettings().setRetrySettings(retrySettings);
+      builder.readRowsSettings().setRetrySettings(retrySettings);
+      builder.mutateRowSettings().setRetrySettings(retrySettings);
+      builder.sampleRowKeysSettings().setRetrySettings(retrySettings);
+    } else {
+
+      builder.bulkMutationsSettings().setSimpleTimeoutNoRetries(Duration.ZERO);
+    }
+
+    builder.setTransportChannelProvider(InstantiatingGrpcChannelProvider.newBuilder()
+        .setChannelsPerCpu(options.getChannelCount())
+        .setMaxInboundMessageSize(retryOptions.getStreamingBufferSize()).build());
```

Comment:
streamingBufferSize is a different concept from inboundMessageSize:
inboundMessageSize is the size of the grpc's internal buffer
streamingBufferSize doesnt appear to be used, but was intended to be used by RowMerger to limit the number of rows to get merged.

I think streamingBufferSize should be removed altogether